Documentation/development.md

```diff
@@ -0,0 +1,84 @@
+# CMO Development
+
+
+## Run CMO Locally
+### Assumptions
+* Should have a cluster running
+* Has setup Go development env
+
+### Build and Run CMO
+
+The `make` target `run-local` builds the `operator` and runs CMO locally
+
+```shell
+export KUBECONFIG=/path/to/kubeconfig/file
+make run-local
+```
+
+### Modify jsonnet
+
+**TL;DR:**
+1. build operator
+1. use `hack/local-cmo.sh` to run the operator
+1. modify assets/yamls
+1. backport the changes to jsonnet
+
+
+Sometimes we may not want to rebuild the operator. E.g. you may need
+to figure out how the stack behaves after applying a change to generated
+kubernetes manifest.  In other words, to modify the YAML in `assets/`
+directory directly without needing to change the `jsonnet/` or
+the `vendored` jsonnet files.
+
+In these cases it is easier to change those yaml files manually, run
+the operator (with those changed files) and later tweak `jsonnet`, than
+to figuring out which parts of jsonnet code you may need to change to
+fix an issue.
+
+It is easier to follow the following workflow
+#### Setup
+1. Start openshift cluster ( e.g. using `cluster-bot` )
+1. `export KUBECONFIG=/path/to/kubeconfig`
+1. Build operator - `make operator`
+
+#### Amend Assets
+1. Amend yaml files in `assets/` directory
+1. Run `hack/local-cmo.sh` (**does not** build the operator)
+1. ☝☝☝ _rinse and repeat_
+
+#### Finally:
+1. Port the changes back to `jsonnet` or vendored `jsonnet`
+
+
+
+## Updating vendored jsonnet code
+
+NOTE: `jb update <repo-url>/<jsonnet-subdir>` doesn't seem to work since it
+does not update the dependencies (at the time of writing)
+
+Steps to update a bundle e.g. `kube-prometheus`:
+
+```shell
+cd jsonnet
+
+
+# edit jsonnetfile.json and remove all but the bundle you want to update
+# E.g. remove everything but *kube-prometheus* from the jsonnetfile.json file
+
+jb update
+cp jsonnetfile.lock.json backup.json
+
+# restore the original jsonnetfile.json file
+git checkout -- jsonnetfile.json # (restore to original)
+
+# apply all changes from backup to lockfile
+vimdiff backup.json  jsonnnetfile.lock.json
```

Comment:
In line 69 you are copying the lock file to backup , so there will not be any changes that need to be moved to the lock file.

Reply:
There should be changes since `jb update` should have changed the lockfile 
Note  that you will need to 

> # edit jsonnetfile.json and remove all but the bundle you want to update
> # E.g. remove everything but *kube-prometheus* from the jsonnetfile.json file

before running the update.